src/pipx/commands/run.py

```diff
@@ -189,14 +188,16 @@ def run(
         # we can't parse this as a package
         package_name = app
 
-    if spec is not None:
-        content = None
-    else:
-        content = maybe_script_content(app, is_path)
+    if python and not Path(python).is_file():
```

Comment:
I guess this check is unnecessary here?
See comment below

Reply:
Hi @dukecat0, nice catch! It looks like I missed cleaning this up. I removed the logic in the run module entirely, since it is handled by main already. 👍